pyfolio/utils.py

```diff
@@ -126,25 +130,38 @@ def load_portfolio_risk_factors(filepath_prefix=None):
         Risk factors timeseries.
     """
 
-    if filepath_prefix is None:
-        import pyfolio
-        filepath = os.path.join(os.path.dirname(pyfolio.__file__), 'data')
-    else:
-        filepath = filepath_prefix
-
-    factors = pd.read_csv(os.path.join(
-        filepath, 'F-F_Research_Data_Factors_daily.csv'), index_col=0)
-    mom = pd.read_csv(os.path.join(
-        filepath, 'daily_mom_factor_returns_fixed_dates2.csv'),
-                      index_col=0, parse_dates=True)
-
-    factors.index = [datetime.fromtimestamp(
-        time.mktime(time.strptime(str(t), "%Y%m%d"))) for t in factors.index]
-
-    five_factors = factors.join(mom)
-    # transform the returns from percent space to raw values (to be consistent
-    # with our portoflio returns values)
-    five_factors = five_factors / 100
+    five_factors = None
+
+    # If it's been more than two days since we updated, redownload CSVs
+    if time.time() - os.path.getmtime('data/factors.h5') > 60*60*24*2:
```

Comment:
I think this could be done more elegantly with a pandas timedelta: http://pandas.pydata.org/pandas-docs/stable/timeseries.html#period